Userland/DevTools/HackStudio/HackStudioWidget.cpp

```diff
@@ -942,6 +943,53 @@ NonnullRefPtr<GUI::Action> HackStudioWidget::create_save_as_action()
     });
 }
 
+ErrorOr<NonnullRefPtr<GUI::Action>> HackStudioWidget::create_toggle_terminal_action()
```

Comment:
I'm not sure about this behavior.

In VsCode, "ctrl + `" pulls up a terminal, and toggling the action hides the entire action bar thingy on the bottom of the window.

In Clion, `alt+F12` pulls up a terminal widget, and pressing the keybind again hides the widget again.

Could we model the behavior more like that? Where pressing the terminal hides the terminal widget altogether until you "request" it again? (but keeping it running in the background, killing any tasks on hide as a bonus).

I feel like changing the focus calls to set_visible() might have the desired effect?



As an aside, you have a typo in the commit message: "focusses" --> "focuses"

Reply:
ahh sure. My initial idea was to hide the entire action tab, but wasn't too sure about it.  I have updated code to hide the entire action tab, instead of just shifting the focus. 😄

Reply:
_"Could we model the behavior more like that? Where pressing the terminal hides the terminal widget altogether until you "request" it again? (but keeping it running in the background, killing any tasks on hide as a bonus)."_  Right now, if we have any task running in the terminal, and we try to `toggle terminal` it hides the action tab (not killing the process running in the terminal. Could you please explain "(but keeping it running in the background, killing any tasks on hide as a bonus)." ? @ADKaster

Reply:
I have attached a screen recording the current behaviour.